blocks/bricks/conv.py

```diff
@@ -98,22 +99,27 @@ def apply(self, input_):
         else:
             W = self.params
 
-        output = conv2d(
-            input_, W, image_shape=self.input_dim, subsample=self.step,
-            border_mode=self.border_mode,
-            filter_shape=((self.num_filters, self.num_channels) +
-                          self.filter_size))
+        image_shape = None
+        if all([x is not None for x in [self.batch_size,
+                                        self.num_channels, self.input_dim]]):
+            image_shape = (self.batch_size, self.num_channels) + self.input_dim
+
+        output = conv2d(input_, W, subsample=self.step,
+                        border_mode=self.border_mode,
+                        image_shape=image_shape,
+                        filter_shape=(self.num_filters, self.num_channels)
+                        + self.filter_size)
         if self.use_bias:
             output += b.dimshuffle('x', 0, 'x', 'x')
         return output
 
     def get_dim(self, name):
         if name == 'input_':
-            return self.input_dim
+            return (self.batch_size, self.num_channels, self.input_dim)
```

Comment:
Yeah, that's correct. Because batch sizes in general can be variable, even though for convolutional networks they might need to be fixed, it probably shouldn't be returned here so that it plays nice with all the other layers.